src/gtfs/mod.rs

```diff
@@ -355,21 +351,106 @@ pub fn read_from_zip<P: AsRef<Path>>(p: P, configuration: Configuration) -> Resu
 /// This method makes it possible to read from a variety of sources like read a GTFS
 /// from the network.
 ///
-/// ```
-// let url = "http://some_url/gtfs.zip";
-// let resp = reqwest::blocking::get(url)?; // or async call
-// let data = std::io::Cursor::new(resp.bytes()?.to_vec());
-// let model = transit_model::gtfs::from_read(data, &url, configuration)?;
+/// ```ignore
+/// let url = "http://some_url/gtfs.zip";
+/// let resp = reqwest::blocking::get(url)?; // or async call
+/// let data = std::io::Cursor::new(resp.bytes()?.to_vec());
+/// let model = transit_model::gtfs::from_zip_reader(data, &url)?;
+/// # Ok::<(), Error>(())
 /// ```
 ///
 /// The `source_name` is needed to have nicer error messages.
-/// The `Configuration` is used to control various parameters during the import.
-pub fn from_read<R>(reader: R, source_name: &str, configuration: Configuration) -> Result<Model>
+pub fn from_zip_reader<R>(reader: R, source_name: &str) -> Result<Model>
 where
     R: std::io::Seek + std::io::Read,
 {
-    let mut file_handler = read_utils::ZipHandler::new(reader, &source_name)?;
-    read(&mut file_handler, configuration)
+    Reader::default().parse_zip_reader(reader, source_name)
+}
+
+/// Imports a `Model` from the
+/// [GTFS](https://gtfs.org/reference/static).
+/// files in the given directory.
+/// This method will try to detect if the input is a zipped archive or not.
+/// If the default file type mechanism is not enough, you can use
+/// [from_zip] or [from_dir].
+pub fn read<P: AsRef<Path>>(p: P) -> Result<Model> {
+    Reader::default().parse(p)
+}
+
+/// Structure to configure the GTFS reading
+#[derive(Default)]
+pub struct Reader {
+    configuration: Configuration,
+}
+
+impl Reader {
+    /// Build a Reader with a custom configuration
+    pub fn new(configuration: Configuration) -> Self {
+        Self { configuration }
+    }
+
+    /// Imports a `Model` from the
+    /// [GTFS](https://gtfs.org/reference/static).
+    /// files in the given directory.
+    /// This method will try to detect if the input is a zipped archive or not.
+    /// If the default file type mechanism is not enough, you can use
+    /// [Reader::parse_zip] or [Reader::parse_dir].
+    pub fn parse(self, path: impl AsRef<Path>) -> Result<Model> {
```

Comment:
If it wasn't already discussed, I'd rather name it `read` (same for other "parse" fn)

Reply:
we chose `parse` over `read` because it would have led to `read_zip_reader` and it was a bit strange (and a first it was even `read_zip_read` :stuck_out_tongue_winking_eye: )